src/routes/pages/urls.ts

```diff
@@ -5,16 +5,33 @@ import {
   createUrl,
   findUrlByShortcode,
   getAllUrlsForUser,
+  PageOptions,
+  PaginationOptions,
 } from '../../controllers/urls'
 
 export const route = Router()
 
 // The entire URLs area is for logged in people only
 route.use(ensureLoggedIn('/login'))
 
+// Pagination middleware
+route.use((req, res, next) => {
+  const LIMIT = 20
+  if (!req.query.page || req.query.page < 0) {
+    req.query.page = 1
+  }
+  req.query.limit = LIMIT
+  next()
+})
+
 route.get('/', async (req, res) => {
-  const urls = await getAllUrlsForUser(req.user)
-  return res.render('pages/urls/index', { urls })
+  const page: PageOptions = {
+    offset: req.query.limit * (req.query.page - 1),
+    limit: req.query.limit,
+  }
+  const { urls, pagination } = await getAllUrlsForUser(req.user, page)
+  const pageList = [...Array(pagination.pageCount).keys()]
```

Comment:
nitpick, but the best method to generate this array is 

```js
Array.from({length: pagination.pageCount}, (v, k) => k+1);
```
https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/from

Reply:
@championswimmer Donzo 😄 🚀